### PR TITLE
Added Radix Sort [Java]

### LINKED ISSUE #530: Radix Sort [Java]
#### This issue is about [RadixSort](https://en.wikipedia.org/wiki/Radix_sort).

- [x] Issue Title is in form `Radix Sort[Java]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by gfredtech**
Fixes #530 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [X] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [X] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [X] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [X] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [X] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

**Comment by gfredtech** [comment]
Thanks, I have now checked the points correctly

**Comment by gfredtech** [comment]
Hello @singhpratyush I am having problems with the Travis build process. It has something to do with how IntelliJ IDE implements newlines but I don't know how I can fix that

**Comment by singhpratyush** [comment]
AFAIK, IntellIJ supports [`.editorconfig`](http://editorconfig.org/). You should be able to resolve such errors using it.

**Comment by singhpratyush** [comment]
@gfredtech: Line endings need to be fixed before merging this. Please ensure the same. Thanks :)

**Comment by gfredtech** [comment]
Line endings fixed now

File: radix_sort/RadixSort.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+public class RadixSort {
+
+    private static void radixSort(int[] array) {
+        int i;
```
Comment (Monal5031) [inline radix_sort/RadixSort.java]: It would be nice if you reduce the scope of variables.

File: radix_sort/RadixSort.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (Monal5031) [inline radix_sort/RadixSort.java]: Please mention the Space and time complexity too.

File: radix_sort/RadixSort.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/**
+ * This Java program implements the Radix sort algorithm
+ * It's a non-comparative based sorting algorithm, hence its worse case time
+ * complexity is O(kn), and space is O(k + n) where k is the bucket size
+ */
+package com.gfred;
```
Comment (Monal5031) [inline radix_sort/RadixSort.java]: Please remove package name.

File: radix_sort/RadixSort.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+public class RadixSort {
+
+    private static void radixSort(int[] array) {
+        protected int i;
```
Comment (Monal5031) [inline radix_sort/RadixSort.java]: @gfredtech By reducing scope here I meant, use `i` directly in the loop itself.

Comment (gfredtech) [reply radix_sort/RadixSort.java]: okay, I'm fixing that

**Comment by Monal5031** [comment]
ping @singhpratyush

**Comment by singhpratyush** [comment]
@gfredtech: Thanks for the code :)

**Pull request merged**